include/sys/sem.h

```diff
@@ -39,6 +39,39 @@ struct sys_sem {
 #endif
 };
 
+/**
+ * @brief Statically define and initialize a sys_sem
+ *
+ * The semaphore can be accessed outside the module where it is defined using:
+ *
+ * @code extern struct sys_sem <name>; @endcode
+ *
+ * Route this to memory domains using K_APP_DMEM().
+ *
+ * @param _name Name of the semaphore.
+ * @param _initial_count Initial semaphore count.
+ * @param _count_limit Maximum permitted semaphore count.
+ */
+#ifdef CONFIG_USERSPACE
+#define SYS_SEM_DEFINE(_name, _initial_count, _count_limit) \
+	struct sys_sem _name = { \
+		.futex = { _initial_count }, \
+		.limit = _count_limit \
+	}
```

Comment:
I think we should also add BUILD_ASSERT here to check init count and count limit as sys_sem_init do.